### DISCUSSION TITLE
Deploy In AWS Lambda

### DISCUSSION THREAD

**Discussion by 0x44454c** [python-telegram-bot/python-telegram-bot#2247] [Q&A]
_**How to deploy Telegram Bot build using PTB in AWS Lambda?**_

**Comment by Bibo-Joshi** [comment]
Please see #534.

**Reply by 0x44454c** [reply]
I tried that thing but since it didn't work I just got frustrated and tried hosting on EC2 Ubuntu instance!
If you don't mind I need a complete solution.😅 Thankyou!

**Reply by Bibo-Joshi** [reply]
You will need to be more specific than "it didn't work". See [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Ask-Right). 😉 
But [this comment](https://github.com/python-telegram-bot/python-telegram-bot/issues/534#issuecomment-282900570) and this [section](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Webhooks#alternative-no-threading) on the Webhooks page already include rather detailed information.

**Reply by 0x44454c** [reply]
Oh thankyou! 
And sorry for the inconveniences!😓
But can you please provide me a example script so that I can implement the same? 

> And also I wasn't able to import this PTB module in AWS Lambda! Why is that??

**Reply by Bibo-Joshi** [reply]
> But can you please provide me a example script so that I can implement the same?

I don't have any except for this [comment](https://github.com/python-telegram-bot/python-telegram-bot/issues/534#issuecomment-489445617) from the issues that I already pointed you to.

> And also I wasn't able to import this PTB module in AWS Lambda! Why is that??

You will need to be more specific than "I wasn't able to". See [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Ask-Right). 😉 Note that it might be an AWS specific problem, which we can't solve here.

**Reply by jojo786** [reply]
I've got a fully working solution at https://github.com/jojo786/TelegramTasweerBot to run on AWS Lambda being AWS API Gateway. Its easy to deploy using AWS SAM.  It uses @erpheus outline from https://github.com/python-telegram-bot/python-telegram-bot/issues/534.

**Reply by jojo786** [reply]
@Bibo-Joshi I added a section to the wiki with details on how to run PTB on AWS. Are you ok with that?

**Reply by Bibo-Joshi** [reply]
TBH, I've already reverted the change … I did this because:

* general info on AWS as host should rather go on the page on [where to host](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Where-to-host-Telegram-Bots)
* your text considered polling, which is not the topic of the `Webhooks` page
* I don't find your repository to be a very good fit as example for how to run a bot on AWS lambda. Ofc by this I don't want to make any judgement about the functionality of your bot! I rather mean that your repository contains a bot with a very specific functionality. For a good example on how to host a bot on AWS lambda, I would expect a kind of "minimal working example" or "skeleton repository". If you look at [this section of the webhooks page](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Webhooks#alternative-no-threading) or [this skeleton repo for hosting on heroku](https://github.com/Bibo-Joshi/ptb-heroku-skeleton) you'll notice that those just provide the basic structure so that for a user it should be easy to grasp how the code can be adapted for their use case 

I hope that you can understand my comments as constructive critisism. Contributions to code & wiki are of course always welcome! We (the dev team) still have to make sure that those contributions fit into the existing resources and improve the material :)

**Reply by jojo786** [reply]
Thanks @Bibo-Joshi. Point taken - I would love to contribute.
1. Can I add just the Webhooks sections back
2. I'm happy to add the other items to  [where to host  ](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Where-to-host-Telegram-Bots). Currently that page just a list of bullet points - can I turn the AWS point into a section?

**Reply by Bibo-Joshi** [reply]
I would suggest to add the info on the different AWS options in the short, informative bullet point style, e.g. along the lines

> * AWS 
>   *  To run a bot using webhooks, set up an AWS API Gateway …
>   * For polling…

If the serverless setup on AWS lambda requires some special setup for PTB, I suggest to add AWS as subsection to the ["no threading"](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Webhooks#alternative-no-threading) section :)

**Comment by jojo786** [comment]
Hi Bibo,

finally got around to this again.

Can I post this on  where to host
<https://github.com/python-telegram-bot/python-telegram-bot/wiki/Where-to-host-Telegram-Bots>
:

   - AWS
      - [Running a Python Telegram Bot on AWS Serverless](
      https://github.com/jojo786/Sample-Python-Telegram-Bot-AWS-Serverless)


Yusuf

**Reply by Bibo-Joshi** [reply]
looks like an elaborate write-up 👍 you can go ahead. please just add a note about which PTB version this was tested with. I'd recommend to also include such in note in the readme of your repo and pin that version in your [requirements file](https://github.com/jojo786/Sample-Python-Telegram-Bot-AWS-Serverless/blob/main/ptb/requirements.txt).

**Comment by jojo786** [comment]
Done.
Included and pinned version 13.11

Yusuf
